src/server.jsx

```diff
@@ -2,30 +2,44 @@ import React from 'react';
 import ReactDOMServer from 'react-dom/server';
 import { Provider } from 'react-redux';
 import { StaticRouter as Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import axios from 'axios';
 
 import configureStore from 'configureStore.js';
 import AppView from 'views/AppView.jsx';
+import Html from 'components/Html/Html.jsx';
 
-export default function (props) {
-  const store = configureStore();
+const authApi = req => (
+  axios.get('http://localhost:3000/api/user', {
+    headers: {
+      Cookie: req.headers.cookie,
+    },
+  })
+    .then(r => r.data)
+    .catch(() => null)
+);
+
+export default async function ({ req, res, context }) {
+  const history = createMemoryHistory({
+    initialEntries: [req.url],
+  });
+  const user = await authApi(req);
+  const store = configureStore(history, { user });
+
+  // We need to wait for the promise to complete to dispatch the actions
   const markup = ReactDOMServer.renderToString(
     <Provider store={store}>
-      <Router location={props.req.url} context={props.context} ><AppView /></Router>
+      <Router location={req.url} context={context} ><AppView /></Router>
     </Provider>,
   );
 
-  return `
-    <html>
-      <head>
-        <link href="/index.css" type="text/css" rel="stylesheet"  />      
-      </head>
-      <body>
-          <div id="app">
-            ${markup}          
-          </div>
-        <script type="text/javascript" src="/index.js"></script>
-      </body>
-    </html>
-  `;
+  if (context.url) {
+    return res.redirect(context.url);
```

Comment:
Context is used by router to tell us that we need to redirect and other relevant data for the router, if something force us to redirect, we send a redirect in the server